numpy/ma/mrecords.py

```diff
@@ -692,6 +693,15 @@ def fromtextfile(fname, delimiter=None, commentchar='#', missingchar='',
 
 
     Ultra simple: the varnames are in the header, one line"""
+    if delimitor is not None:
+        # NumPy 1.22.0, 2021-09-22
+        warnings.warn('The "delimitor" keyword argument of '
+                      'numpy.ma.mrecords.fromtextfile() is deprecated '
+                      'since NumPy 1.22.0, use "delimiter" instead.',
+                      DeprecationWarning, stacklevel=2)
+        if delimiter is None:
+            delimiter = delimitor
```

Comment:
If both are provided I think you should just error:
```suggestion
    if delimitor is not None:
        if delimiter is not None:
            raise TypeError("fromtextfile got multiple values for argument delimiter")
        # NumPy 1.22.0, 2021-09-22
        warnings.warn('The "delimitor" keyword argument of '
                      'numpy.ma.mrecords.fromtextfile() is deprecated '
                      'since NumPy 1.22.0, use "delimiter" instead.',
                      DeprecationWarning, stacklevel=2)
        delimiter = delimitor
```

Reply:
Yes, makes absolute sense.

Reply:
Do we want the warning **in addition** to the exception? Might help debug this very improbable case...

Reply:
I could move the `raise TypeError` after the `warnings.warn`.